sdk/core/azure_core/src/error/error_response.rs

```diff
@@ -108,19 +108,43 @@ struct ErrorDetailsInternal<'a> {
     message: Option<&'a str>,
 }
 
+/// Options for customizing the behavior of `check_success`.
+#[derive(Debug, Default)]
+pub struct CheckSuccessOptions {
+    /// A list of HTTP status codes that should be considered successful.
+    ///
+    /// If this list is empty, any 2xx status code is considered successful.
+    pub success_statuses: &'static [u16],
```

Comment:
The error response pertains to http? Why don't we use the http::status struct?
https://docs.rs/http/latest/http/status/struct.StatusCode.html